webapp/channels/src/components/user_groups_modal/user_groups_list/user_groups_list.tsx

```diff
@@ -77,97 +109,136 @@ const UserGroupsList = React.forwardRef((props: Props, ref?: React.Ref<HTMLDivEl
+                                            goToViewGroupModal(group);
+                                        }}
+                                        icon={<i className='icon-account-multiple-outline'/>}
+                                        text={Utils.localizeMessage('user_groups_modal.viewGroup', 'View Group')}
+                                        disabled={false}
+                                    />
+                                </Menu.Group>
+                                <Menu.Group>
+                                    <Menu.ItemAction
+                                        show={groupPermissionsMap[group.id].can_delete}
+                                        onClick={() => {
+                                            archiveGroup(group.id);
+                                        }}
+                                        icon={<i className='icon-archive-outline'/>}
+                                        text={Utils.localizeMessage('user_groups_modal.archiveGroup', 'Archive Group')}
+                                        disabled={false}
+                                        isDangerous={true}
+                                    />
+                                    <Menu.ItemAction
+                                        show={groupPermissionsMap[group.id].can_restore}
+                                        onClick={() => {
+                                            restoreGroup(group.id);
+                                        }}
+                                        icon={<i className='icon-restore'/>}
+                                        text={Utils.localizeMessage('user_groups_modal.restoreGroup', 'Restore Group')}
+                                        disabled={false}
+                                    />
+                                </Menu.Group>
+                            </Menu>
+                        </MenuWrapper>
                     </div>
-                );
-            })}
-            {
-                (loading) &&
-                <LoadingScreen/>
-            }
+                </div>
+            );
+        }
+        if (loading) {
+            return <LoadingScreen/>;
+        }
+        return null;
+    };
+
+    return (
+        <div
+            className='user-groups-modal__content user-groups-list'
+            style={{overflow: overflowState}}
+        >
+            <InfiniteLoader
+                ref={infiniteLoaderRef}
+                isItemLoaded={isItemLoaded}
+                itemCount={100000}
+                loadMoreItems={loadMoreItems}
+            >
+                {({onItemsRendered, ref}) => (
+                    <VariableSizeList
+                        itemCount={itemCount}
+                        onItemsRendered={onItemsRendered}
+                        ref={ref}
+                        itemSize={() => 52}
+                        height={groups.length >= 8 ? 416 : groups.length * 52}
```

Comment:
To prevent downwards cutoff

<img width="618" alt="CleanShot 2023-08-22 at 15 21 22@2x" src="https://github.com/mattermost/mattermost/assets/11724372/c988a427-8302-4955-88b3-d0dc4f559259">
<img width="618" alt="CleanShot 2023-08-22 at 15 36 03@2x" src="https://github.com/mattermost/mattermost/assets/11724372/c56e143b-89de-4ee4-961f-90ca9c1e6ff1">


```suggestion
                        height={groups.length >= 8 ? 416 : Math.max(groups.length, 3) * 52}
```

Reply:
fixed https://github.com/mattermost/mattermost/pull/22597/commits/35aef46f275f1bf8f5612629097e346b2c3d209f